src/server/game/Entities/Player/Player.cpp

```diff
@@ -2510,6 +2510,18 @@ void Player::RemoveFromWorld()
     }
 }
 
+bool Player::IsImmunedToSpellEffect(SpellInfo const* spellInfo, uint32 index) const
+{
+    // players are immune to taunt (the aura and the spell effect).
+    if (uint32 aura = spellInfo->Effects[index].ApplyAuraName)
+        if (aura == SPELL_AURA_MOD_TAUNT)
+            return true;
+    if (spellInfo->Effects[index].Effect == SPELL_EFFECT_ATTACK_ME)
+        return true;
```

Comment:
you could use spellinfo helpers `SpellEffectInfo::IsEffect` and `SpellEffectInfo::IsAura`